provisioning/device/index.d.ts

```diff
@@ -1,7 +1,8 @@
 // Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
-export { Authentication, TransportHandlers, ResponseCallback, ClientConfiguration, DeviceConfiguration } from './lib/interfaces';
-export { ClientStateMachine } from './lib/client_state_machine';
+export { ProvisioningAuthentication, PollingTransportHandlers, ProvisioningTransportOptions, ProvisioningDeviceConfiguration, ProvisioningTransportHandlersBase, ProvisioningTransportHandlersX509 } from './lib/interfaces';
```

Comment:
>ProvisioningAuthentication [](start = 9, length = 26)

nit: that's a very long line that is probably worth cutting

Reply:
also do we use all of these in other packages?

---
In reply to: [151319163](https://github.com/Azure/azure-iot-sdk-node/pull/154#discussion_r151319163) [](ancestors = 151319163)

Reply:
Yes we do.  Most of these have to do with passing information between the device client and the transport.

---
In reply to: [151319210](https://github.com/Azure/azure-iot-sdk-node/pull/154#discussion_r151319210) [](ancestors = 151319210,151319163)